### PR TITLE
core/core: add scr.prompt_sec variable to show section name in prompt

### PR DESCRIPTION
**Pull request by ret2libc**
You can obtain something like:
```
[0.__text:main + 13]>
```

### CODE REVIEW

File: libr/core/core.c [hunk 1]
```diff
@@ -1049,6 +1048,13 @@ static int prompt_flag (RCore *r, char *s, size_t maxlen) {
 	return R_TRUE;
 }
 
+static void prompt_sec(RCore *r, char *s, size_t maxlen) {
+	RIOSection *sec = r_io_section_vget (r->io, r->offset);
```
Comment (jvoisin) [inline libr/core/core.c]: + `const`

Comment (ret2libc) [inline libr/core/core.c]: thanks!

**Comment by jvoisin** [comment]
Why are you using `int` instead of `unsigned int` ?

**Comment by ret2libc** [comment]
Where?

File: libr/core/core.c [hunk 2]
```diff
@@ -1036,11 +1036,10 @@ static int prompt_flag (RCore *r, char *s, size_t maxlen) {
 	if (!f) return R_FALSE;
 
 	if (f->offset != r->offset) {
-		snprintf (s, maxlen, "%s + %d",
-			f->name, (int)(r->offset - f->offset));
+		snprintf (s, maxlen, "%s + %d", f->name,
+			(int)(r->offset - f->offset));
```
Comment (jvoisin) [inline libr/core/core.c]: Here; why `int` and not `unsigned int` ?

Comment (ret2libc) [inline libr/core/core.c]: Right, I just took the code from the cmd_flag.c file where it was like that

**Comment by ret2libc** [comment]
fixed

**Comment by jvoisin** [comment]
LGTM :water_buffalo:

**Comment by radare** [comment]
i would suggest scr.promptsect to keep consistencey with scr.promptflag and such, what if sect and flag are set? what if we want to get always the offset? imho the offset should be always displayed but maybe we can add promptoff

**Comment by ret2libc** [comment]
My mistake in the commit message. The variable is called promptsec. If sect
and flag are both set the output is like what I wrote in the first message.

The offset is always displayed when you use prompt flag, except when the
offset is zero. IMO it's OK like this. Having "+ 0"  would be redundant imho
Il 11/ago/2015 20:48, "radare" <notifications@github.com> ha scritto:

**Comment by ret2libc** [comment]
I moved prompt on the heap and refactored the code.

**Comment by radare** [comment]
lgtm!

**Pull request closed without merging**